src/commands/targeting/disable.ts

```diff
@@ -23,47 +25,26 @@ export default class DisableTargeting extends Base {
 
         await this.requireProject()
 
-        let responses
-
-        const feature = args['feature']
-        const environment = args['environment']
-
-        if (flags.headless && (!feature || !environment)) {
-            throw new Error('In headless mode, both the feature and environment are required')
-        }
-
-        if (feature) {
-            responses = { featureKey: feature }
-        } else {
-            const userSelectedFeature = await inquirer.prompt(
-                [featurePrompt],
-                {
-                    token: this.authToken,
-                    projectKey: this.projectKey
-                }
-            )
-            responses = { featureKey: userSelectedFeature.feature }
-        }
-
-        if (environment) {
-            responses = { ...responses, environmentKey: environment }
-        } else {
-            const userSelectedEnv = await inquirer.prompt<EnvironmentPromptResult>(
-                [environmentPrompt],
-                {
-                    token: this.authToken,
-                    projectKey: this.projectKey
-                }
-            )
-            responses = { ...responses, environmentKey: userSelectedEnv.environment._id }
-        }
-
-        const disableTargetingForFeatureAndEnvironment = await disableTargeting(
+        const responses = await getFeatureAndEnvironmentKeyFromArgs(
+            this.authToken, 
+            this.projectKey, 
+            args, 
+            flags,
+        )
+        const updatedTargeting = await disableTargeting(
             this.authToken,
             this.projectKey,
-            responses.featureKey,
-            responses.environmentKey
-        )
-        return this.writer.showResults(disableTargetingForFeatureAndEnvironment)
+            responses.featureKey as string,
+            responses.environmentKey as string
+        ) 
+    
+        if (flags.headless) {
+            this.writer.showResults(updatedTargeting)
+        } else {
+            // TODO: reuse the data fetched for the prompts
+            const environments = await fetchEnvironments(this.authToken, this.projectKey)
```

Comment:
we could just fetch the related environment by key, since we're only updating one